src/Control/Varying/Core.hs

```diff
@@ -83,15 +99,15 @@ import Debug.Trace
 --------------------------------------------------------------------------------
 -- | Lift a pure computation into a stream.
 var :: Applicative m => (a -> b) -> VarT m a b
-var f = VarT $ \a -> pure (f a, var f)
+var f = VarT $ \(!a) -> pure (f a, var f)
```

Comment:
I'm not quite sure these `!a`s make sense. Did you see a difference in benchmarking?

If the function `a` is passed to does not inspect its argument, then unnecessary work is done. And if it does, then it will force it itself!

It might make sense to seq `f a` instead (and all such first elements of return values of `VarT` things), basically as pair was defined as `data (,) a b = (,) !a b`. Note that the second element is already in WHNF, so no need to bang it.

This would, however, dirty up all the rest of the code a lot more than bang patterns does :D